### DISCUSSION TITLE
Policy matching on queue-type

### DISCUSSION THREAD

**Discussion by gomoripeti** [rabbitmq/rabbitmq-server#6217] [Ideas] [answered]
Currently it is only possible to restrict what a policy is applied to by
- queues or exchanges or all
- regex on name
- indirectly if using definition keys which are listed as unsupported capabilities for some queue types

It would be more explicit (and maybe more future proof) to allow explicitly selecting which queue type we intend a policy to be applied to.

For example one could specify a "default for quorum queues" policy with a conservative delivery-limit value (as quorum queues don't handle that well infinite redelivery). Today that policy won't be applied to other queue types but there was mention of implementing delivery-limit for classic queues. Maybe then we would like to apply a different policy with a different limit for CQs as they have different memory/disk characteristics.

Could be implemented with a new optional policy parameter `queue-type` which would allow the same values as `x-queue-type`. (I think it is easier to implement and more backwards compatible than extending the allowed values of `apply-to`)

What do you think?

**Comment by michaelklishin** [comment]
That's a decent idea that was discussed by the core team. We would consider a PR for sure.

**Comment by gomoripeti** [comment]
great, thanks for the positive response.

**Comment by mkuratczyk** [comment, marked as answer]
Completed in https://github.com/rabbitmq/rabbitmq-server/pull/7601. Will be available in 3.12

**Comment by gomoripeti** [comment]
that's great. thank you!
